src/snowflake/snowpark/column.py

```diff
@@ -235,14 +238,27 @@ def __init__(
         self._ast = ast
 
         if expr2 is not None:
-            # TODO: Find correct entities in AST for df_aliasing
             if isinstance(expr1, str) and isinstance(expr2, str):
                 if expr2 == "*":
                     self._expression = Star([], df_alias=expr1)
                 else:
                     self._expression = UnresolvedAttribute(
                         quote_name(expr2), df_alias=expr1
                     )
+
```

Comment:
What would the alias field here be? is this for `expr.as_(alias)`? Can we maybe add an example what would need to get captured here?

Reply:
The alias field would be something that the user provides to the `DataFrame.alias(self, name: str)` method.

The Snowpark API then allows users to create columns that reference this DataFrame instance by its alias using a line like `col(<df_alias>, <column_name>)`

In the IR we'll need the alias to resolve which `DataFrame` instance the user is referring to, but there might be a better way to go about enabling this functionality.

Reply:
prob. would  be good to include your explanation as comment here